### ISSUE TITLE
How is PyTorch tested before its release?

### ISSUE THREAD

**Issue by HLH13297997663** [pytorch/pytorch#137590]
Hello， Could you please explain how comprehensive testing is conducted when adding new features to PyTorch? Is there end-to-end testing involved?
```[tasklist]
### Tasks
- [ ] Does PyTorch conduct end-to-end testing before releasing a new version? For example, are there tests for model training and loss alignment?
```

**Comment by HLH13297997663** [comment]
Does PyTorch conduct end-to-end testing before releasing a new version? For example, are there tests for model training and loss alignment?

**Comment by malfet** [comment]
Thank you for creating the issue, we do run some end to end testing and we do run entire set of benchmarks,  see https://hud.pytorch.org/torchbench/userbenchmark for example

Please use https://discuss.pytorch.org/ to ask those types of questions as issues should primarily be used for reporting issue and asking for new features. If you have a feature to suggest or bug to report, please do not hesitate to open a new issue

**Comment by atalman** [comment]
hi @HLH13297997663 There are also smoke test validations for all release binaries: https://github.com/pytorch/builder/blob/main/.github/workflows/validate-binaries.yml

Also validation checklist: https://github.com/pytorch/pytorch/issues/137492

Also we ask for our partners help to do some testing. You are also welcome to do testing on your end. 
Please consult for installation details: https://dev-discuss.pytorch.org/t/pytorch-release-2-5-0-final-rc-is-available/2514

**Comment by HLH13297997663** [comment]
> hi @HLH13297997663 There are also smoke test validations for all release binaries: https://github.com/pytorch/builder/blob/main/.github/workflows/validate-binaries.yml
> 
> Also validation checklist: #137492
> 
> Also we ask for our partners help to do some testing. You are also welcome to do testing on your end. Please consult for installation details: https://dev-discuss.pytorch.org/t/pytorch-release-2-5-0-final-rc-is-available/2514

Thank you for your reply. May I ask:
(1) What model will be chosen for training smoke testing?
(2) What are the passing criteria for end-to-end training tests? Is loss alignment performed?
